examples/http-client-server/server/server.bal

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'int as ints;
+import ballerina/tcp;
+import ballerina/regex;
+import ballerina/io;
+
+enum stateMachine {
+    WAITING, RECEIVING_BODY, RECEIVED_BODY
+}
+
+enum Method {
+    POST
+}
+
+public enum HttpVersion {
+    HTTP_1_1,
+    HTTP_2
+}
+
+type Request record {
+    map<string> headers;
+    string? body = ();
+    string HttpVersion = HTTP_1_1;
+    string Method = POST;
+    string path;
+};
+
+type Response record {
+    map<string> headers;
+    string? body = ();
+    string status;
+    string HttpVersion;
+};
+
+service on new tcp:Listener(3000) {
+    remote function onConnect(tcp:Caller caller) returns tcp:ConnectionService {
+        return new EchoHttpService();
+    }
+}
+
+service class EchoHttpService {
+
+    private string status = WAITING;
+    private string? payload = ();
+    private map<string> headersMap = {};
+    private int payloadLength = 0;
+    private string httpVersion = "HTTP/1.1";
+    private string httpStatus = "200 Ok";
+
+    remote isolated function onBytes(tcp:Caller caller, readonly & byte[] data) returns tcp:Error? {
+        string|error request = string:fromBytes(data);
+        lock {
+            self.httpStatus = request is error ? "404 Bad Request" : "200 Ok";
+        }
+        if (request is error) {
```

Comment:
We can write it as follows right,

```ballerina
if request is error
```

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-tcp/pull/379/commits/1599b6dfa51ac438fb4098f1237142abffb5d78e